extra/sequences/extras/extras.factor

```diff
@@ -559,6 +559,18 @@ PRIVATE>
 : infimum-by* ( ... seq quot: ( ... elt -- ... x ) -- ... i elt )
     [ before? ] select-by* ; inline
 
+: ?supremum ( seq/f -- elt/f )
```

Comment:
You don't need the ``dup [ ... ] when`` because ``f`` is defined as an empty sequence.

So this would work:

```factor
: ?supremum ( seq -- elt/f )
    [ f ] [ [ ] [ 2dup and [ max ] [ dupd ? ] if ] map-reduce ] if-empty ;
```